File: solutions/student-attendance-record-i/student-attendance-record-i.go
// @Title: Student Attendance Record I
// @Author: colinjxc
// @Date: 2018-12-15T21:53:46+08:00
// @URL: https://leetcode-cn.com/problems/student-attendance-record-i


func checkRecord(s string) bool {
    l, a := 0, 0
    for i := 0; i < len(s); i++ {
        if (s[i] == 'L') {
            l += 1
        } else if (s[i] == 'A') {
            l = 0;
            a += 1
        } else {
            l = 0;
        }
        
        if l > 2 || a > 1 {
            return false
        }
    }
    return true
}

File: solutions/student-attendance-record-i/student-attendance-record-i.cpp
// @Title: Student Attendance Record I
// @Author: colinjxc
// @Date: 2018-12-15T19:44:59+08:00
// @URL: https://leetcode-cn.com/problems/student-attendance-record-i


class Solution {
public:
    bool checkRecord(string s) {
        int record[256] = {0, };

        for (int i = 0; i < s.size(); i++) {
            char c = s[i];
            record[c] += 1;
            if (c == 'L') {
                if (record[c] > 2) return false;
                if (c != s[i + 1]) record[c] = 0;
            } else if (c == 'A') {
                if (record[c] > 1) return false;
            }
        }
        return true;
    }
};
